docs/creating-contracts.md

```diff
@@ -0,0 +1,79 @@
+While developing AEAs, users frequently face the need of interacting with smart contracts in a third-party decentralized ledger. To achieve this, developers must create <a href="../contract">Contract</a> packages: components that provide wrappers around the API or ABI of the target smart contract.
+
+In this guide, we will learn how to develop our own contract package to interface with an ERC20 token. We will use the WETH contract deployed on Ethereum [at this address](https://etherscan.io/token/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2#readContract).
+
+## Step-by-step instructions
+
+Before starting this guide, ensure that you have done one of the following things:
+
+- You have cloned our [developer template](https://github.com/valory-xyz/dev-template). Once setup, it will generate a virtual environment with Open AEA installed, an empty local registry, some useful tools for checking packages and dummy tests.
+
+- You have gone through the [quickstart](./quickstart.md) to verify that your machine satisfies the framework requirements and that you have followed the setup instructions so you have the Open AEA framework and the Ethereum plugin installed.
+
+Now, let's create our contract package:
+
+1. Download the WETH contract ABI [here](https://etherscan.io/token/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2#code): scroll down to the **Contract ABI** section and click the **copy** button on the right. Then, create a new file called `IERC20.json` and paste the content you have just copied there.
+
+2. Initialize the IPFS registry:
+```bash
+autonomy init --reset --author john_doe --remote --ipfs --ipfs-node "/dns/registry.autonolas.tech/tcp/443/https"
+```
+
+3. Scaffold the ERC20 contract:
+```bash
+aea scaffold --to-local-registry contract ERC20Contract /path/to/IERC20.json
+```
+You'll find the contract in the local registry at `packages/john_doe/contracts/ERC20Contract`.
+
+4. Open the contract at `packages/john_doe/contracts/ERC20Contract/contract.py` and add the following imports at the top of the file:
+```python
+from aea_ledger_ethereum import EthereumApi
+from typing import Optional
+```
+
+    Also add a method to retrieve balances to the `ERC20Contract` class
+```python
+    @classmethod
+    def balance_of(
+        cls, ledger_api: EthereumApi, contract_address: str, owner_address: str
+    ) -> Optional[JSONLike]:
+        """Gets an account's balance."""
+        contract_instance = cls.get_instance(ledger_api, contract_address)
+
+        return ledger_api.contract_method_call(
+            contract_instance=contract_instance,
+            method_name="balanceOf",
+            owner=owner_address,
+        )
+```
```

Comment:
This neatly demonstrates why the PR #262 does not achieve it's purpose. The ledger connection can automatically resolve messages specifying the target `balanceOf` to this method call. No single line of code has to be written by the dev.

Reply:
Yes, but at some point devs might need to have custom code in their implementation, so I think we should leave this and also add the new option.